cruise-control-core/src/main/java/com/linkedin/cruisecontrol/monitor/sampling/MetricSample.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.cruisecontrol.monitor.sampling;
+
+import com.linkedin.cruisecontrol.resource.Resource;
+import java.util.HashMap;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class MetricSample<E> {
+  private static final Logger LOG = LoggerFactory.getLogger(MetricSample.class);
+  protected final E _entity;
+  protected final Map<Resource, Double> _metrics;
```

Comment:
This can just be double[Resource.values().length] which will save lots of space.  null values can be represented as Double.NaN

Reply:
The metric samples are short lived objects. They will be discarded once we aggregate the data into the metric sample aggregator.

Reply:
That's fine, but it's lots of extra garbage as well.  Number-of-entities x (number-of-metrics + 1) vs number-of-entities.